main.tf

```
@@ -1,23 +1,29 @@
 locals {
-  enabled = module.this.enabled
+  enabled                            = module.this.enabled
+  fargate_profile_enabled            = var.fargate_profile_enabled && local.enabled
+  fargate_pod_execution_role_enabled = var.fargate_pod_execution_role_enabled && local.enabled
 
-  tags = merge(
+  tags = try(length(var.cluster_name), 0) == 0 ? module.this.tags : merge(
     module.this.tags,
     {
       "kubernetes.io/cluster/${var.cluster_name}" = "owned"
     }
   )
 
+  fargate_pod_execution_role_name = var.fargate_pod_execution_role_name != null ? var.fargate_pod_execution_role_name : var.fargate_profile_iam_role_name
+
   fargate_profile_name = var.fargate_profile_name != null ? var.fargate_profile_name : module.fargate_profile_label.id
 
-  fargate_profile_iam_role_name = var.fargate_profile_iam_role_name != null ? var.fargate_profile_iam_role_name : (
-  "${module.role_label.id}${var.iam_role_kubernetes_namespace_delimiter}${var.kubernetes_namespace}")
+  fargate_profile_iam_role_name = local.fargate_pod_execution_role_name != null ? local.fargate_pod_execution_role_name : (
+    var.fargate_profile_enabled ? "${module.role_label.id}${var.iam_role_kubernetes_namespace_delimiter}${var.kubernetes_namespace}" :
+  module.role_label.id)
```

I think this would be less painful like this
```suggestion
  fargate_profile_concatenated = "${module.role_label.id}${var.iam_role_kubernetes_namespace_delimiter}${var.kubernetes_namespace}"
  default_fargate_profile_role_name = var.fargate_profile_enabled ? local.fargate_profile_concatenated :  module.role_label.id
  fargate_profile_iam_role_name = coalesce(local.fargate_pod_execution_role_name, local.default_fargate_profile_role_name)
```

---

Problem is:

```hcl
fargate_profile_concatenated = "${module.role_label.id}${var.iam_role_kubernetes_namespace_delimiter}${var.kubernetes_namespace}"
```

fails with Error if `var.kubernetes_namespace` is null, which it could be if user is only provisioning the Pod Execution Role.

---

okay, I wouldn't see why that couldn't be coalesced as well.
So:
```
fargate_profile_concatenated = "${module.role_label.id}${var.iam_role_kubernetes_namespace_delimiter}${coalesce(var.kubernetes_namespace,"")}"
```
or if any number of the elements could be null, then use:
```
fargate_profile_concatenated = join("", compact([module.role_label.id,var.iam_role_kubernetes_namespace_delimiter,var.kubernetes_namespace]))
```

---

To be clear, I'm just offering options to avoid a nested conditional.

---

`coalesce` fails when all the arguments are empty. The advantage of the ternary operator in Terraform is that the unused expressions are not evaluated and need not be valid.